From 83dd4cbb41138e87fdbeead90cb18686896c2369 Mon Sep 17 00:00:00 2001
From: Richard Guy Briggs <rgb@redhat.com>
Date: Wed, 1 Oct 2014 21:35:59 -0400
Subject: [kernel] audit: __audit_syscall_entry - ignore arch arg and call syscall_get_arch() directly

Message-id: <36439f28599adfaaff5ac74882913a25d0cf3384.1412192166.git.rgb@redhat.com>
Patchwork-id: 96863
O-Subject: [RHEL7 PATCH V2 6/9] audit: __audit_syscall_entry: ignore arch arg and call syscall_get_arch() directly
Bugzilla: 1053921
RH-Acked-by: Eric Paris <eparis@redhat.com>
RH-Acked-by: Paul Moore <pmoore@redhat.com>

Since every arch should have syscall_get_arch() defined, stop using the
function argument and just collect this ourselves.  We do not drop the
argument as fixing some code paths (in assembly) to not pass this first
argument is non-trivial.  The argument will be dropped when that is
fixed.

Upstream-commit: 4a99854
Signed-off-by: Richard Guy Briggs <rgb@redhat.com>
Signed-off-by: Eric Paris <eparis@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/kernel/auditsc.c b/kernel/auditsc.c
index 9c149eb..40f49c6 100644
--- a/kernel/auditsc.c
+++ b/kernel/auditsc.c
@@ -1467,7 +1467,7 @@ void __audit_syscall_entry(int arch, int major,
  if (!audit_enabled)
   return;
 
- context->arch     = arch;
+ context->arch     = syscall_get_arch();
  context->major      = major;
  context->argv[0]    = a1;
  context->argv[1]    = a2;
-- 
1.7.1